src/components/cl/doca_urom/cl_doca_urom.h

```diff
@@ -0,0 +1,105 @@
+/**
+ * Copyright (c) 2020, NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+ *
+ * See file LICENSE for terms.
+ */
+
+#ifndef UCC_CL_DOCA_UROM_H_
+#define UCC_CL_DOCA_UROM_H_
+
+#include "components/cl/ucc_cl.h"
+#include "components/cl/ucc_cl_log.h"
+#include "components/tl/ucc_tl.h"
+#include "coll_score/ucc_coll_score.h"
+#include "utils/ucc_mpool.h"
+
+#include <doca_dev.h>
+#include <doca_urom.h>
+#include <doca_pe.h>
+#include <doca_ctx.h>
+#include <doca_buf.h>
+
+#include <worker_ucc.h>
+#include <common.h>
+#include <urom_common.h>
+#include <urom_ucc.h>
+
+#ifndef UCC_CL_DOCA_UROM_DEFAULT_SCORE
+#define UCC_CL_DOCA_UROM_DEFAULT_SCORE 100
+#endif
+
+#define UCC_CL_DOCA_UROM_ADDR_MAX_LEN  1024
+#define UCC_CL_DOCA_UROM_MAX_TEAMS     16
+
+typedef struct ucc_cl_doca_urom_iface {
+    ucc_cl_iface_t super;
+} ucc_cl_doca_urom_iface_t;
+
+// Extern iface should follow the pattern: ucc_cl_<cl_name>
+extern ucc_cl_doca_urom_iface_t ucc_cl_doca_urom;
+
+typedef struct ucc_cl_doca_urom_lib_config {
+    ucc_cl_lib_config_t super;
+    char *device;
+    char *plugin_name;
+    int doca_log_level;
+} ucc_cl_doca_urom_lib_config_t;
+
+typedef struct ucc_cl_doca_urom_context_config {
+    ucc_cl_context_config_t super;
+    ucs_config_names_array_t plugin_envs;
+} ucc_cl_doca_urom_context_config_t;
+
+typedef struct urom_ctx {
+    struct doca_urom_service                   *urom_service;
+    struct doca_urom_worker                    *urom_worker;
+    struct doca_urom_domain                    *urom_domain;
+    struct doca_pe                             *urom_pe;
+    const struct doca_urom_service_plugin_info *ucc_info;
+    void                                       *urom_worker_addr;
+    size_t                                      urom_worker_len;
+    uint64_t                                    worker_id;
+    void                                       *urom_ucc_context;
+    ucc_rank_t                                  ctx_rank;
+} urom_ctx_t;
+
+typedef struct ucc_cl_doca_urom_lib {
+    ucc_cl_lib_t                  super;
+    ucc_cl_doca_urom_lib_config_t cfg;
+    urom_ctx_t                    urom_ctx;
```

Comment:
It probably should be renamed to `ucc_cl_doca_urom` something, all the more since it is exposed in the header.
Also I would put this field at the ctx level.